scraper_service/main.py

```diff
@@ -1,79 +1,90 @@
-    filename = f'shackcam/{timestamp_str}' + phone + '.jpg'
+    # Receive message from upstream service
+    event_message = event['Records'][0]['Sns']['Message']
+    phone_number = event_message['phone_number'].split("B")[-1]
 
-    # Create message and send to predict_service_trigger
-    # on topic "triggerPredict"
-    outbound_message = {'image_id': filename,
-                        'phone': phone}
+    timestamp = datetime.now(timezone('EST'))
+    timestamp_str = timestamp.strftime('%Y_%m_%d_%H%M')
+    filename = f'shackcam/{timestamp_str}.jpg'
 
-    sns = boto3.client('sns')
-    response = sns.publish(
-        TopicArn='arn:aws:sns:us-east-1:245636212397:triggerPredict',
-        Message=json.dumps({'default': json.dumps(outbound_message)}),
-        MessageStructure='json'
-    )
-    print(response)
+    outbound_message = {'filename': filename, 'phone_number': phone_number}
 
-    # Try 3 times to load an image
+    # Try 3 times to scrape an image
     fail_count = 0
-    for x in range(0, 2):
+    for _ in range(0, 2):
         try:
+            # Scrape
             r = make_request()
-        except ValueError as err:
-            print(err)
-            fail_count += 1
-            sleep(1)
-            pass
+            r.raw.decode_content = True
 
-    # Upload image to S3
-    if fail_count >= 2:
-        # If shack cam is failing, just send an old image to S3
-        s3 = boto3.resource('s3')
-        copy_source = {'Bucket': 'deepshack', 'Key': 'samples/s_test.jpg'}
-        bucket = s3.Bucket('deepshack')
-        bucket.copy(copy_source, filename)
-        # write_to_dynamo(filename, phone, 0)
+            # Upload to S3
+            session = boto3.Session()
+            s3 = session.resource('s3')
+            bucket = s3.Bucket(S3_BUCKET)
+            bucket.upload_fileobj(r.raw, filename)
 
-    else:
-        # Scrape the image if shack cam is working
-        session = boto3.Session()
-        s3 = session.resource('s3')
-        bucket_name = 'deepshack'
-        bucket = s3.Bucket(bucket_name)
+            # Publish message to downstream services
+            publish_message(outbound_message, PREDICT_TOPIC_ARN)
+            outbound_message['body'] = (
+                "Good news! DeepShack is on your way."
+                "Image was scraped from ShackCam and saved on AWS S3."
+                "predict_trigger_service on Lambda will be called next"
+            )
+            publish_message(outbound_message, SMS_TOPIC_ARN)
 
-        r.raw.decode_content = True
-        bucket.upload_fileobj(r.raw, filename)
-        # write_to_dynamo(filename, phone, 0)
+        except ValueError as err:
+            logging.info(err)
+            fail_count += 1
+            sleep(1)
 
-    return outbound_message
+        # Publish failed message to downstream services
+        outbound_message['body'] = (
+                "Bad news:( Image was not successfully scraped from ShackCam."
```

Comment:
Make sure to add a space after the period, also before the ":("

Reply:
All set